autonat.go

```diff
@@ -94,10 +100,15 @@ func New(ctx context.Context, h host.Host, options ...Option) (AutoNAT, error) {
 	if conf.forceReachability {
 		emitReachabilityChanged.Emit(event.EvtLocalReachabilityChanged{Reachability: conf.reachability})
 
+		if conf.reachability == network.ReachabilityPublic && len(conf.dialAddrs) == 0 {
+			return nil, xerrors.New("dial addrs must be configured if forced reachability is public")
+		}
+
```

Comment:
I disagree. "forced public" just means that the user _knows_ that the node will be reachable so we should behave as such. That doesn't mean the user will statically configure addresses ahead of time.